pkg/model/hornet/message_id.go

```diff
@@ -0,0 +1,148 @@
+package hornet
+
+import (
+	"encoding/hex"
+	"fmt"
+	"sort"
+
+	iotago "github.com/iotaledger/iota.go/v2"
+)
+
+// MessageID is the ID of a Message.
+type MessageID []byte
+
+// MessageIDs is a slice of MessageID.
+type MessageIDs []MessageID
+
+// ToHex converts the MessageID to its hex representation.
+func (m MessageID) ToHex() string {
+	return hex.EncodeToString(m)
+}
+
+// ToArray converts the MessageID to an array.
+func (m MessageID) ToArray() iotago.MessageID {
+	var messageID iotago.MessageID
+	copy(messageID[:], m)
+	return messageID
+}
+
+// ToMapKey converts the MessageID to a string that can be used as a map key.
+func (m MessageID) ToMapKey() string {
+	return string(m)
+}
+
+// GetNullMessageID returns the ID of the genesis message.
+func GetNullMessageID() MessageID {
+	nullMessageID := make(MessageID, 32)
+	return nullMessageID
+}
+
+// MessageIDFromHex creates a MessageID from a hex string representation.
+func MessageIDFromHex(hexString string) (MessageID, error) {
+
+	b, err := hex.DecodeString(hexString)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(b) != iotago.MessageIDLength {
+		return nil, fmt.Errorf("unknown messageID length (%d)", len(b))
+	}
+
+	return MessageID(b), nil
+}
+
+// MessageIDFromMapKey creates a MessageID from a map key representation.
+func MessageIDFromMapKey(mapKey string) MessageID {
+	if len(mapKey) != iotago.MessageIDLength {
+		panic(fmt.Sprintf("unknown messageID length (%d)", len(mapKey)))
+	}
+
+	return MessageID(mapKey)
+}
+
+// MessageIDFromSlice creates a MessageID from a byte slice.
+func MessageIDFromSlice(b []byte) MessageID {
+
+	if len(b) != iotago.MessageIDLength {
+		panic(fmt.Sprintf("unknown messageID length (%d)", len(b)))
+	}
+
+	return MessageID(b)
+}
+
+// MessageIDFromArray creates a MessageID from a byte array.
+func MessageIDFromArray(b iotago.MessageID) MessageID {
+	return MessageID(b[:])
+}
+
+// ToHex converts the MessageIDs to their hex string representation.
+func (m MessageIDs) ToHex() []string {
+	var results []string
+	for _, msgID := range m {
+		results = append(results, msgID.ToHex())
+	}
+	return results
+}
+
+// ToSliceOfSlices converts the MessageIDs to a slice of byte slices.
+func (m MessageIDs) ToSliceOfSlices() [][]byte {
+	var results [][]byte
+	for _, msgID := range m {
+		// this is necessary, otherwise we create a pointer to the loop variable
+		tmp := msgID
+		results = append(results, tmp)
```

Comment:
```suggestion
		results = append(results, msgID)
```
Declaring a new variable and thus copying the value of `msgID` has no effect as msgID is a `[]byte`. In go the value of a slice is its _header_ which contains a pointer to the underlying array. (This is different for arrays, where an assignment copies all its values)